In [1]:
from pandas import *

In [18]:
curr = read_csv('Curret_fullMV.csv', encoding='latin', low_memory=False)

In [30]:
curr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30270 entries, 0 to 30269
Columns: 106 entries, URL to Unnamed: 105
dtypes: float64(1), object(105)
memory usage: 24.5+ MB


Look how many unused columns you have:

In [35]:
cols = curr.columns
cols[-100:]

Index(['2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14',
       '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26',
       '27', '28', '29', '30', '31', '32', '33', '34', '35', '36', '37', '38',
       '39', '40', '41', '42', '43', '44', 'Unnamed: 49', 'Unnamed: 50',
       'Unnamed: 51', 'Unnamed: 52', 'Unnamed: 53', 'Unnamed: 54',
       'Unnamed: 55', 'Unnamed: 56', 'Unnamed: 57', 'Unnamed: 58',
       'Unnamed: 59', 'Unnamed: 60', 'Unnamed: 61', 'Unnamed: 62',
       'Unnamed: 63', 'Unnamed: 64', 'Unnamed: 65', 'Unnamed: 66',
       'Unnamed: 67', 'Unnamed: 68', 'Unnamed: 69', 'Unnamed: 70',
       'Unnamed: 71', 'Unnamed: 72', 'Unnamed: 73', 'Unnamed: 74',
       'Unnamed: 75', 'Unnamed: 76', 'Unnamed: 77', 'Unnamed: 78',
       'Unnamed: 79', 'Unnamed: 80', 'Unnamed: 81', 'Unnamed: 82',
       'Unnamed: 83', 'Unnamed: 84', 'Unnamed: 85', 'Unnamed: 86',
       'Unnamed: 87', 'Unnamed: 88', 'Unnamed: 89', 'Unnamed: 90',
       'Unnamed: 91', 

In [25]:
unamed_col = [col for col in curr.columns if 'Unnamed' in col]
len(unamed_col)

57

In [31]:
curr[unamed_col].head()

Unnamed: 49 Unnamed: 50 Unnamed: 51 Unnamed: 52 Unnamed: 53 Unnamed: 54  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

  Unnamed: 55 Unnamed: 56 Unnamed: 57 Unnamed: 58     ...      Unnamed: 96  \
0         NaN         NaN         NaN         NaN     ...              NaN   
1         NaN         NaN         NaN         NaN     ...              NaN   
2         NaN         NaN         NaN         NaN     ...              NaN   
3         NaN         NaN         NaN         NaN     ...              NaN   
4         NaN         NaN         NaN         NaN     ...              NaN   

  Unnamed: 97 Unnamed: 98 Unnamed: 99 Unnamed: 100 Unnamed: 101 Unnamed: 102  \
0         NaN         NaN         NaN          NaN          NaN          NaN   
1         NaN         NaN         NaN          NaN          NaN          NaN   
2         NaN         NaN         NaN          NaN          NaN          NaN   
3         NaN         NaN         NaN          NaN          NaN          NaN   
4         NaN         NaN         NaN          NaN          NaN          NaN   

  Unnamed: 103 Unnamed: 104 Unnamed: 105  
0          NaN          NaN          NaN  
1          NaN          NaN          NaN  
2          NaN          NaN          NaN  
3          NaN          NaN          NaN  
4          NaN          NaN          NaN  

[5 rows x 57 columns]

## Columns with only null values

your dataframed size is 30270 rows. amount of null values in each column:

In [40]:
curr.isnull().sum()

URL                   23507
CANONICAL NAME            0
LONG DISPLAY NAME         0
SHORT DISPLAY NAME        0
FULL NAME                 1
1                      9890
2                     24390
3                     27619
4                     28788
5                     29398
6                     29705
7                     29906
8                     30019
9                     30072
10                    30123
11                    30161
12                    30175
13                    30195
14                    30209
15                    30224
16                    30231
17                    30233
18                    30235
19                    30235
20                    30239
21                    30243
22                    30250
23                    30253
24                    30255
25                    30257
                      ...  
Unnamed: 76           30269
Unnamed: 77           30269
Unnamed: 78           30269
Unnamed: 79           30269
Unnamed: 80         

So you have dozens of rows containing only one value. Looking at those:

In [51]:
curr[~curr['Unnamed: 105'].isnull()][-5:]

URL CANONICAL NAME LONG DISPLAY NAME SHORT DISPLAY NAME    FULL NAME  \
10150  NaN    FinnMedi Oy       FinnMedi Oy        FinnMedi Oy  FinnMedi Oy   

                1    2    3    4    5          ...          Unnamed: 96  \
10150  @@FinnMedi  NaN  NaN  NaN  NaN          ...                  NaN   

      Unnamed: 97 Unnamed: 98 Unnamed: 99 Unnamed: 100  \
10150         NaN         NaN         NaN          NaN   

                              Unnamed: 101                       Unnamed: 102  \
10150  Scientific Products (Asia Pacific)   Fisher Worldwide Distribution SPV   

                                 Unnamed: 103                Unnamed: 104  \
10150  Fisher Worldwide Gene Distribution SPV  Fisher WWD Holdings L.L.C.   

               Unnamed: 105  
10150  New FS Holdings Inc.  

[1 rows x 106 columns]

In [49]:
curr[~curr['Unnamed: 99'].isnull()]

URL     CANONICAL NAME  \
10206  https://www.fishersci.com/us/en/catalog/search...  Fisher Scientific   

       LONG DISPLAY NAME SHORT DISPLAY NAME                       FULL NAME  \
10206  Fisher Scientific  Fisher Scientific  Fisher Scientific Company, LLC   

                       1                    2                3  \
10206  fisher scientific  fisher sci company   Fisher Chemical   

                                             4                 5     ...       \
10206  Consultores Fisher Scientific Chile Ltd  FisherScientific     ...        

                 Unnamed: 96       Unnamed: 97   Unnamed: 98  \
10206  FSUK Holdings Limited  FSWH Company LLC  FSWH II C.V.   

                        Unnamed: 99 Unnamed: 100 Unnamed: 101 Unnamed: 102  \
10206  Scientific Products (India)           NaN          NaN          NaN   

      Unnamed: 103 Unnamed: 104 Unnamed: 105  
10206          NaN          NaN          NaN  

[1 rows x 106 columns]

Tons of garbage. To clean, google "drop null columns pandas" and see the dropna method.

In [56]:
to_cut_out = read_csv('WantToCutOut_fromOldMV.csv')

In [57]:
to_cut_out.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22794 entries, 0 to 22793
Data columns (total 9 columns):
source                22794 non-null object
URL                   248 non-null object
CANONICAL NAME        22794 non-null object
LONG DISPLAY NAME     22794 non-null object
SHORT DISPLAY NAME    22794 non-null object
FULL NAME             22794 non-null object
1                     12996 non-null object
2                     224 non-null object
3                     15 non-null object
dtypes: object(9)
memory usage: 1.6+ MB


In [59]:
to_keep = read_csv('WantToKeepMostClean_fromOldMV.csv', sep = '\t')
to_keep.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10367 entries, 0 to 10366
Data columns (total 50 columns):
sorce                 10367 non-null object
URL                   6554 non-null object
CANONICAL NAME        10367 non-null object
LONG DISPLAY NAME     10367 non-null object
SHORT DISPLAY NAME    10367 non-null object
FULL NAME             10367 non-null object
1                     6778 non-null object
2                     3854 non-null object
3                     2395 non-null object
4                     1425 non-null object
5                     840 non-null object
6                     546 non-null object
7                     341 non-null object
8                     229 non-null object
9                     181 non-null object
10                    138 non-null object
11                    104 non-null object
12                    86 non-null object
13                    67 non-null object
14                    57 non-null object
15                    43 non-null objec

# Shapes (num rows and columns)

In [61]:
curr.shape

(30270, 106)

In [62]:
to_cut_out.shape

(22794, 9)

In [63]:
to_keep.shape

(10367, 50)

# Finding what's common and what's not

In [65]:
join_col = 'CANONICAL NAME'
curr[join_col].head()

0                [iito]
1       1 A Pharma GmbH
2          10X Genomics
3           121 Bio LLC
4    13therapeutics Inc
Name: CANONICAL NAME, dtype: object

## To keep

Number of Names in "to keep" that are also in curr

In [75]:
to_keep[join_col].isin(curr[join_col]).sum(), len(to_keep)

(10179, 10367)

In [72]:
to_keep['isInCurr'] = to_keep[join_col].isin(curr[join_col])

So some companies in "to_keep" are not in curr. These companies are:

In [77]:
to_keep.query('isInCurr == False')[join_col].head()

17            Fisher BioImage ApS
20               Coris BioConcept
71    Advanced Targeting Systems 
73             AdvaTec Analytics 
74                        AE Adam
Name: CANONICAL NAME, dtype: object

## to cut out

In [78]:
to_cut_out[join_col].isin(curr[join_col]).sum(), len(to_cut_out)

(20165, 22794)

In [74]:
to_cut_out['isInCurr'] = to_cut_out[join_col].isin(curr[join_col])

In [79]:
to_cut_out.query('isInCurr == False')[join_col].head()

0          DePuy
1     13advisory
3         2cureX
7      4P-Pharma
10      A-Filter
Name: CANONICAL NAME, dtype: object

## current file

In curr and in "to keep"

In [83]:
curr[join_col].isin(to_keep[join_col]).sum(), len(curr)

(10173, 30270)

In curr and in "to_cut_out":

In [84]:
curr[join_col].isin(to_cut_out[join_col]).sum(), len(curr)

(18806, 30270)

In [85]:
curr['toCutOut'] = curr[join_col].isin(to_cut_out[join_col])
curr['toCutOut'].sum()

18806

In [86]:
curr['toKeep'] = curr[join_col].isin(to_keep[join_col])
curr['toKeep'].sum()

10173

Compenies in current that are in to keep or to cut out:

In [89]:
len(curr.query('toKeep or toCutOut')), len(curr)

(28963, 30270)

In [90]:
30270 - 28963

1307

so current has only 1307 companies which are not in "to keep" or "to cut out". Remeber also that there are a few hundreds of companies which are in "to keep" and "to cut out" which are not in curr. This might be a descrepancy you need to look at.

# Seperating stuff

now you can take what ever you like from it for example:

In [92]:
cutted_out_from_current = curr.query('toCutOut')
curr_without_cut_out_piece = curr.query('not toCutOut')
cutted_out_from_current.shape, curr_without_cut_out_piece.shape

((18806, 108), (11464, 108))

In [93]:
cutted_out_from_current.to_csv('cutted_out_from_current.csv', index = False)

In [94]:
curr_without_cut_out_piece.to_csv('curr_without_cut_out_piece.csv', index = False)

Or:

In [96]:
in_toKeep_but_not_in_curr = to_keep.query('not isInCurr')

In [97]:
in_toKeep_but_not_in_curr.to_csv('in_toKeep_but_not_in_curr.csv', index = False)